openml/tasks/task.py

```diff
@@ -161,27 +305,69 @@ def __init__(self, task_id, task_type_id, task_type, data_set_id,
             evaluation_measure=evaluation_measure,
             target_name=target_name,
             data_splits_url=data_splits_url,
+            estimation_procedure_id=estimation_procedure_id
         )
 
 
 class OpenMLClusteringTask(OpenMLTask):
-    def __init__(self, task_id, task_type_id, task_type, data_set_id,
-                 evaluation_measure, number_of_clusters=None):
+    def __init__(
+            self,
+            task_type_id: Union[int, str],
+            task_type: str,
+            data_set_id: int,
+            evaluation_measure: Optional[str] = None,
+            target_name: Optional[str] = None,
+            task_id: Optional[Union[int, str]] = None,
+            estimation_procedure_id: int = 17,
+    ):
         super(OpenMLClusteringTask, self).__init__(
             task_id=task_id,
             task_type_id=task_type_id,
             task_type=task_type,
             data_set_id=data_set_id,
             evaluation_measure=evaluation_measure,
         )
-        self.number_of_clusters = number_of_clusters
+        self.target_name = target_name
+        self.estimation_procedure_id = estimation_procedure_id
+
+    def _to_dict(self) -> Dict[str, OrderedDict]:
+
+        task_container = super(OpenMLClusteringTask, self)._to_dict()
+        task_dict = task_container['oml:task_inputs']
+
+        task_dict['oml:input'].append(
```

Comment:
Nice observation. It also made me realise that the code for some of the tasks can be incomplete since when the class is downloaded from the server it has more information concerning the estimation procedure.